KGP.TicketApp.Repositories/ClientRepository.cs

```diff
@@ -0,0 +1,30 @@
+﻿using KGP.TicketApp.Contracts;
+using KGP.TicketApp.Model.Database;
+using KGP.TicketApp.Model.Database.Tables;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using static KGP.TicketApp.Model.Database.Tables.User;
+
+namespace KGP.TicketApp.Repositories
+{
+    public class ClientRepository : RepositoryBase<Client>, IClientRepository
+    {
+        #region Constructors
+
+        public ClientRepository(DatabaseContext context) : base(context) {}
+
+        #endregion
+
+        #region Interface methods
+
+        public Client? FindUserByEmail(string email)
+        {
+            return DatabaseContext.Set<Client>().FirstOrDefault(user => user.Email == email);
```

Comment:
kminię jak zrobić, żeby nie duplikować kodu tu i w repozytorium organizatorów. Może zrobić konkretną klasę UserRepository<T>, która dziedziczy po RepositoryBase<T>. Tylko wtedy trzeba będzie zrobić kowarancję/kontrawariancję (nie pamiętam które xD). Ale to jakieś trudne i w sumie może się okazać że to za dużo pierdzielenia, jak ci się nie chce to można to olać xD

Reply:
w sumie można zrobić klasę abstrakcyjną dać tam zaimplementować i odziedziczyć po niej do clienta i organizera